modules/javafx.controls/src/main/java/javafx/scene/control/TableView.java

```diff
@@ -427,6 +427,19 @@
         }
     };
 
+    public static final Callback<TableView.ResizeFeatures, Boolean> CONSTRAINED_RESIZE_POLICY_ALL_COLUMNS =
+        ConstrainedColumnResize.forTable(ConstrainedColumnResize.ResizeMode.AUTO_RESIZE_ALL_COLUMNS);
+
+    public static final Callback<TableView.ResizeFeatures, Boolean> CONSTRAINED_RESIZE_POLICY_LAST_COLUMN =
+        ConstrainedColumnResize.forTable(ConstrainedColumnResize.ResizeMode.AUTO_RESIZE_LAST_COLUMN);
+
+    public static final Callback<TableView.ResizeFeatures, Boolean> CONSTRAINED_RESIZE_POLICY_NEXT_COLUMN =
+        ConstrainedColumnResize.forTable(ConstrainedColumnResize.ResizeMode.AUTO_RESIZE_NEXT_COLUMN);
+
+    // TODO to be renamed to CONSTRAINED_RESIZE_POLICY
+    public static final Callback<TableView.ResizeFeatures, Boolean> CONSTRAINED_RESIZE_POLICY_SUBSEQUENT_COLUMNS =
```

Comment:
Instead of renaming we can also deprecate the 'old' `CONSTRAINED_RESIZE_POLICY` ?

Reply:
Good question.

We could deprecate the old policy, although leaving it as is also leaves all its issues in place (see JDK-8292810).  There is a subtle behavioral difference between the old CONSTRAINED_RESIZE_POLICY and CONSTRAINED_RESIZE_POLICY_SUBSEQUENT_COLUMNS that people might notice and may not like.  The difference is that the old policy works similar to CONSTRAINED_RESIZE_POLICY_LAST_COLUMN until the last column cannot be resized, then it resizes one before it.

Another option is that we could add another policy which will work exactly (minus bugs) as the old one.  What do you think?

Reply:
Try tests/manual/controls/ATableViewResizeTester.java to see the behavior.

Reply:
> We could deprecate the old policy, although leaving it as is also leaves all its issues in place (see JDK-8292810). There is a subtle behavioral difference between the old CONSTRAINED_RESIZE_POLICY and CONSTRAINED_RESIZE_POLICY_SUBSEQUENT_COLUMNS that people might notice and may not like. The difference is that the old policy works similar to CONSTRAINED_RESIZE_POLICY_LAST_COLUMN until the last column cannot be resized, then it resizes one before it.
> 
> Another option is that we could add another policy which will work exactly (minus bugs) as the old one. What do you think?

I like that all new constrained resize policies have a more describing name and I think we should stick to it, therefore I'm in favor of deprecating the 'old', more generic named resize policy.
Regarding the small behaviour